app/Main.hs

```diff
@@ -96,49 +96,95 @@ writePackageFile =
   . packageConfigToJSON
 
 data PackageInfo = PackageInfo
-  { repo         :: Text
+  { repo         :: Repo
   , version      :: Text
   , dependencies :: [PackageName]
   } deriving (Show, Eq, Generic, Aeson.FromJSON, Aeson.ToJSON)
 
 type PackageSet = Map.Map PackageName PackageInfo
 
-cloneShallow
+newtype Repo = Repo { unRepo :: Text } deriving (Show, Eq)
+
+instance Aeson.FromJSON Repo where
+  parseJSON = fmap Repo . Aeson.parseJSON
+
+
+instance Aeson.ToJSON Repo where
+  toJSON = Aeson.toJSON . unRepo
+
+
+data CloneTarget = CloneTag Text
+                 | CloneSHA Text
+                 deriving (Show)
+
+
+parseCloneTarget
   :: Text
+  -> Either Text CloneTarget
+parseCloneTarget t =
+  if T.null remainder
+     then Right (CloneTag t)
+     else case T.toLower schemeName of
+      "sha" -> Right (CloneSHA withoutScheme)
+      "tag" -> Right (CloneTag withoutScheme)
+      _ -> Left ("Invalid scheme. Expected sha: | tag: but got " <> schemeName)
```

Comment:
I guess we should probably allow using tags by default and warn about it. Thoughts? cc @kRITZCREEK

Reply:
I'm a bit confused. I could see adding a case for `"" -> Right (CloneTag withoutScheme)`. But if they affirmatively provided a scheme that's some unknown, it seems like we should still error.

Reply:
I should've read all of the code instead of just this bit :D I see now that we only hit this case if the target contained a `:` in the first place.